docs/advanced/metadata.md

```diff
@@ -133,68 +133,68 @@ We are almost there, but we still don't have the "treatment" metadata captured i
 
 We can call the `getParent()` method on each of the paths like so:
 
-```groovy linenums="1"
-map { id, reads ->
-    reads.collect { it.getParent() }
+```groovy linenums="3" hl_lines="2"
+.map { id, reads ->
+    reads.collect { read -> read.getParent() }
 }
 ```
 
 If we want to call a set method on every item in a Collection, Groovy provides this convenient "spread dot" notation:
 
-```groovy linenums="1"
-map { id, reads ->
+```groovy linenums="3" hl_lines="2"
+.map { id, reads ->
     reads*.getParent()
 }
 ```
 
 This returns another Path object, but we only want the name of the last directory, so we need to call `.getName()` method on each of these Paths. We can use the spread-dot notation again:
 
-```groovy linenums="1"
-map { id, reads ->
+```groovy linenums="3" hl_lines="2"
+.map { id, reads ->
     reads*.getParent()*.getName()
 }
 ```
 
 The last piece of Groovy sugar is to note that methods with `get` and `set` prefixes can be called with a property-style notation, converting `getParent()` to `parent` and `getName()` to `name`:
 
-```groovy linenums="1"
-map { id, reads ->
+```groovy linenums="3" hl_lines="2"
+.map { id, reads ->
     reads*.parent*.name
 }
 ```
 
 If we wanted to remove the "treatment" prefix, we can combine this new notation with the "minus" method which we used earlier in the aliased `-` form.
 
-```groovy linenums="1"
-map { id, reads ->
+```groovy linenums="3" hl_lines="2"
+.map { id, reads ->
     reads*.parent*.name*.minus(~/treatment/)
 }
 ```
 
 In this particular example, we know ahead of time that the treatments must be the same because of the way the `fromFilePairs` method gathers pairs, but we'll continue for the sake of the demonstration. Our final `map` closure might look like:
 
-```groovy linenums="1"
+```groovy linenums="1" hl_lines="5"
 workflow {
     Channel.fromFilePairs("data/reads/*/*_R{1,2}.fastq.gz")
-    | map { id, reads ->
-        (sample, replicate, type) = id.tokenize("_")
-        (treatmentFwd, treatmentRev) = reads*.parent*.name*.minus(~/treatment/)
-        meta = [
-            sample:sample,
-            replicate:replicate,
-            type:type,
-            treatmentFwd:treatmentFwd,
-            treatmentRev:treatmentRev,
-        ]
-        [meta, reads]
-    }
-    | view
+        .map { id, reads ->
+            def (sample, replicate, type) = id.tokenize("_")
+            def (treatmentFwd, treatmentRev) = reads*.parent*.name*.minus(~/treatment/)
+            def meta = [
+                sample:sample,
+                replicate:replicate,
+                type:type,
+                treatmentFwd:treatmentFwd,
+                treatmentRev:treatmentRev,
+            ]
+            [meta, reads]
```

Comment:
The future type checker will like it if you create tuples with the `tuple()` function instead of a list